ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/StatsUpdater.java

```diff
@@ -86,15 +89,33 @@ public void gatherStats(CompactionInfo ci, HiveConf hiveConf,
             } else {
                 sb.append(" noscan");
             }
-            LOG.info(ci + ": running '" + sb + "'");
-            if (compactionQueueName != null && compactionQueueName.length() > 0) {
-                conf.set(TezConfiguration.TEZ_QUEUE_NAME, compactionQueueName);
-            }
+
+            LOG.info("{}: running '{}'", ci, sb);
             SessionState sessionState = DriverUtils.setUpAndStartSessionState(conf, userName);
             DriverUtils.runOnDriver(conf, sessionState, sb.toString());
         } catch (Throwable t) {
-            LOG.error(ci + ": gatherStats(" + ci.dbname + "," + ci.tableName + "," + ci.partName +
-                    ") failed due to: " + t.getMessage(), t);
+          LOG.error("{}: gatherStats({},{},{}) failed due to: {}",
+                  ci, ci.dbname, ci.tableName, ci.partName, t.getMessage(), t);
         }
     }
+
+    HiveConf setUpDriverSession(
```

Comment:
should it be private?

Reply:
It is called from unit tests testing this method.